src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 200) {
+    this.health = health;
+    this.name = 'Animal';
+  }
+
+  dead(healthPoints) {
+    this.health += healthPoints;
+
+    if (this.health < 100) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health);
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, 100) {
+    this.hidden = 'hidden';
```

Comment:
[SYNTAX ERROR] The constructor should accept 'name' as a parameter and pass it to the super constructor. The health should be set to 100 by default, not passed as a parameter. Also, 'hidden' should be initialized to false, not 'hidden'.